Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/SphereComponentMode.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ComponentModes/SphereComponentMode.h>
+
+#include <AzToolsFramework/ComponentModes/ShapeTranslationOffsetViewportEdit.h>
+#include <AzToolsFramework/Manipulators/RadiusManipulatorRequestBus.h>
+#include <AzToolsFramework/Manipulators/ShapeManipulatorRequestBus.h>
+#include <AzToolsFramework/ViewportSelection/EditorSelectionUtil.h>
+
+namespace AzToolsFramework
+{
+    void InstallSphereViewportEditFunctions(SphereViewportEdit* sphereViewportEdit, const AZ::EntityComponentIdPair& entityComponentIdPair)
+    {
+        auto getSphereRadius = [entityComponentIdPair]()
+        {
+            float radius = 0.5f;
+            RadiusManipulatorRequestBus::EventResult(radius, entityComponentIdPair, &RadiusManipulatorRequestBus::Events::GetRadius);
+            return radius;
+        };
+        auto getRotationOffset = [entityComponentIdPair]()
+        {
+            AZ::Quaternion rotationOffset = AZ::Quaternion::CreateIdentity();
+            ShapeManipulatorRequestBus::EventResult(
+                rotationOffset, entityComponentIdPair, &ShapeManipulatorRequestBus::Events::GetRotationOffset);
+            return rotationOffset;
+        };
+        auto setSphereRadius = [entityComponentIdPair](float radius)
+        {
+            RadiusManipulatorRequestBus::Event(entityComponentIdPair, &RadiusManipulatorRequestBus::Events::SetRadius, radius);
+        };
+        sphereViewportEdit->InstallGetSphereRadius(AZStd::move(getSphereRadius));
+        sphereViewportEdit->InstallGetRotationOffset(AZStd::move(getRotationOffset));
+        sphereViewportEdit->InstallSetSphereRadius(AZStd::move(setSphereRadius));
+    }
+
+    AZ_CLASS_ALLOCATOR_IMPL(SphereComponentMode, AZ::SystemAllocator, 0)
+
+        void SphereComponentMode::Reflect(AZ::ReflectContext* context)
```

Comment:
nit: Looks like the macro above has thrown off the indentation here

Reply:
Oh yeah, that happened a bunch of times, guess I missed this one.